ivy/functional/frontends/tensorflow/nn.py

```diff
@@ -123,3 +123,7 @@ def batch_normalization(x, mean, variance, offset, scale, variance_epsilon, name
     return x * ivy.astype(inv, x.dtype, copy=False) + ivy.astype(
         offset - mean * inv if offset is not None else -mean * inv, x.dtype
     )
+
+
+def avg_pool1d(input, ksize, strides, padding, data_format='NWC', name=None):
+    return ivy.avg_pool1d(input, ksize, strides, padding, data_format=data_format)
```

Comment:
I don't think there's an `avg_pool1d` function in Ivy's functional API, so either you'll have to implement the `tensorflow` `avg_pool1d` frontend as a composition of existing Ivy functions or extend Ivy's functional API by adding `ivy.avg_pool1d` in a separate PR referring to the [docs](https://lets-unify.ai/ivy/contributing/4_open_tasks.html#ivy-api-extensions)